app/models/agents/webhook_agent.rb

```diff
@@ -46,10 +49,30 @@ def receive_web_request(params, method, format)
       secret = params.delete('secret')
       return ["Not Authorized", 401] unless secret == interpolated['secret']
 
-      #check the verbs
+      # check the verbs
       verbs = (interpolated['verbs'] || 'post').split(/,/).map { |x| x.strip.downcase }.select { |x| x.present? }
       return ["Please use #{verbs.join('/').upcase} requests only", 401] unless verbs.include?(method)
 
+      # check the reCAPTCHA response if required
+      if recaptcha_secret = interpolated['recaptcha_secret'].presence
+        recaptcha_response = params.delete('g-recaptcha-response') or
+          return ["Not Authorized", 401]
+
+        begin
+          response = faraday.post('https://www.google.com/recaptcha/api/siteverify', {
+                                    secret: recaptcha_secret,
+                                    response: recaptcha_response,
+                                    # remoteip: request.env['REMOTE_ADDR'],
```

Comment:
REMOTE_ATTR may or may not hold the actual IP address of a client depending on if a reverse proxy / load balancer is configured properly.  I think I can make it an option.